Automattic/uCrop
# [Publishing] Add Publish to S3 using Buildkite

App Infra Request: pdnsEh-1RV-p2

Depends On:
- [x] Security related inquiry/discussion on p1727693975599849-slack-C02KLTL3MKM.
- [x] [BuildkiteCI#526](https://github.com/Automattic/buildkite-ci/pull/526)
- [x] Configure `.mobile-secrets` (create deployment key & environmental variables),  and `GitHub` (add deployment key).

### Description

This PR adds publishing to our S3 Maven using Buildkite:

### Testing Instructions

`TODO`:
- Wait for this [BuildkiteCI#526](https://github.com/Automattic/buildkite-ci/pull/526) PR to get deployed/merged.
- Trigger a CI build on this PR and wait for the library to be published on our S3 Maven.
- Use a project like [Gravatar-SDK-Android](https://github.com/Automattic/Gravatar-SDK-Android) to update the group from `com.github.yalantis` to `com.automattic`, while using a `<PR>-<COMMIT_SHA>` combination as its test version.
- Build the test project and verify that any `uCrop` related functionality is still working as expected.
